### PR TITLE
Sort Order

### LINKED ISSUE #1021: Sorting does not set an arrow/triangle (such as ▼/▲) to indicate the sorting order in the popover menu
Since the GTK4 port, we don't have a treeview header area anymore, but we have a sorting popover menubutton.

The state and direction of sorting should be reflected within that menu, on startup and when it is changed by the user.

### PR DESCRIPTION
**Pull request by diegogangl**
This PR adds a couple of buttons to reverse the sorting order (ascending or descending), also converts the sorting options to stateful actions which gives us radio buttons

closes #1021

### CODE REVIEW

File: GTG/core/sorters.py [hunk 1]
```diff
@@ -31,14 +31,46 @@ def unwrap(row, expected_type):
...
 
-    def __init__(self):
-        super(TaskTitleSorter, self).__init__()
+    @reverse.setter
+    def reverse(self, value: bool) -> None:
+        self._reverse = value
+        self.changed(Gtk.SorterChange.INVERTED)
 
 
+    def reversible_compare(self, first, second) -> Gtk.Ordering:
```
Comment (SqAtx) [inline GTG/core/sorters.py:52]: An alternative is to have the equal test first:
```
        if first == second:
            return Gtk.Ordering.EQUAL

        if first < second:
            if self._reverse:
                return Gtk.Ordering.LARGER
            else:
                return Gtk.Ordering.SMALLER

        if self._reverse:
            return Gtk.Ordering.SMALLER
        else:
            return Gtk.Ordering.LARGER
```

The entire thing looks a bit clunky to me, but is apparently the GTK way to solve this problem, so LGTM.

**Comment by diegogangl** [comment]
Used stateful actions for the sorting menu, it's looking a lot better now

Screenshot_20240914_190814

**Review by SqAtx — APPROVED** [review]
Well you updated the review while I was looking at it ^^'

It looks a lot better now!  The UI is clear and usable.

I do think we can iterate again on the UI later.  I don't have concrete suggestions right now, but it seems to me that there is potential to make it a bit more compact / require fewer clicks. But yeah, very happy with how it turned out :)

File: GTG/core/sorters.py (hunk 1 above)
Comment (diegogangl) [reply GTG/core/sorters.py:52]: Ah that's a bit cleaner. Thanks! But yeah, I couldn't find a less clunky way of doing this

**Pull request merged**